admin/main.go

```diff
@@ -184,6 +185,178 @@ func verifyIDToken(app *firebase.App, idToken string) *auth.Token {
+	// [START create_user]
+	u, err := client.CreateUser(context.Background(),
+		(&auth.UserToCreate{}).
+			Email("user@example.com").
+			EmailVerified(false).
+			PhoneNumber("+15555550100").
+			Password("secretPassword").
+			DisplayName("John Doe").
+			PhotoURL("http://www.example.com/12345678/photo.png").
+			Disabled(false))
+
+	if err != nil {
+		log.Fatalf("error creating user: %v\n", err)
+	}
+	log.Printf("Successfully created user: %v\n", u)
+	// [END create_user]
+	return u
+}
+
+func createUserWUID(ctx context.Context, client *auth.Client) *auth.UserRecord {
+	uid := "something"
+	// [START create_user_with_uid]
+	u, err := client.CreateUser(context.Background(),
+		(&auth.UserToCreate{}).UID(uid).Email("user@example.com").PhoneNumber("+15555550100"))
+	// Alternatively
+	params := auth.UserToCreate{} // also possible: 	var params auth.UserToCreate
+	u2, err := client.CreateUser(context.Background(),
+		params.UID(uid).Email("user@example.com").PhoneNumber("+15555550100"))
+
+	if err != nil {
+		log.Fatalf("error creating user: %v\n", err)
+	}
+	log.Printf("Successfully created user: %v\n", u)
+	// [END create_user_with_uid]
+	return u2
+}
+
+func updateUser(ctx context.Context, client *auth.Client) (*auth.UserRecord, *auth.UserRecord) {
+	uid := "d"
+	// [START update_user]
+	u, err := client.UpdateUser(context.Background(), uid,
+		(&auth.UserToUpdate{}).
+			Email("user@example.com").
+			EmailVerified(true).
+			PhoneNumber("+15555550100").
+			Password("newPassword").
+			DisplayName("John Doe").
+			PhotoURL("http://www.example.com/12345678/photo.png").
+			Disabled(true))
+	// Alternatively
+	params := auth.UserToUpdate{} // also possible: 	var params auth.UserToUpdate
+	u2, err := client.UpdateUser(context.Background(), uid,
+		params.Email("user@example.com").PhoneNumber("+15555550100"))
+
+	if err != nil {
+		log.Fatalf("error updating user: %v\n", err)
+	}
+	log.Printf("Successfully updated user: %v\n", u)
+	// [END update_user]
+	return u, u2
+}
+
+func deleteUser(ctx context.Context, client *auth.Client) {
+	uid := "d"
+	// [START delete_user]
+	err := client.DeleteUser(context.Background(), uid)
+	if err != nil {
+		log.Fatalf("error deleting user: %v\n", err)
+	}
+	log.Printf("Successfully deleted user: %s\n", uid)
+	// [END delete_user]
+}
+
+func customClaims(ctx context.Context, client *auth.Client) {
+	uid := "uid"
+	// [START custom_claims]
+
+	// erase all existing custom claims
+	err := client.SetCustomUserClaims(context.Background(), uid, nil)
+	if err != nil {
+		log.Fatalf("error removing custom claims %v", err)
+	}
+	// add custom claims
+	err = client.SetCustomUserClaims(context.Background(), uid, map[string]interface{}{"2custom": "2claims"})
+	if err != nil {
+		log.Fatalf("error setting custom claims %v", err)
+	}
+	// Alternatively
+	_, err = client.UpdateUser(context.Background(), uid,
+		(&auth.UserToUpdate{}).CustomClaims(map[string]interface{}{"2custom": "2claims"}))
+
+	// [END custom_claims]
+
+}
```

Comment:
Need more sample snippets to shadow the Node.js samples in https://firebase-dot-devsite.googleplex.com/docs/auth/admin/custom-claims

Specifically:
* Accessing claims via decoded ID tokens
* Accessing claims via a call to `GetUser()`
* Defining roles via backend script
* Incrementally updating claims